samza-sql/src/main/java/org/apache/samza/sql/fn/BuildOutputRecordUdf.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql.fn;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang.Validate;
+import org.apache.samza.config.Config;
+import org.apache.samza.sql.SamzaSqlRelRecord;
+import org.apache.samza.sql.udfs.ScalarUdf;
+
+
+/**
+ * UDF that builds a SamzaSqlRelMessage.SamzaSqlRelRecord with given list of key value pairs.
+ * Useful if you need to populate fields for a Kafka message.
+ *
+ * For example, given args = {k1, v1, k2, v2},
+ * it returns a SamzaSqlRelRecord with fieldNames={k1, k2} and fieldValues={v1, v2},
+ * where v1 or v2 can be any Object, including SamzaSqlRelRecord when you want to set nested field.
+ *
+ * Consider the below nested output schema:
+ * {
+ *   field1,
+ *   field2:{
+ *     field21,
+ *     field22
+ *   },
+ *   field3:{
+ *     field31:{
+ *       field311,
+ *       field312
+ *     },
+ *     field32
+ *   }
+ * };
+ * It could be built in the select statement as:
+ * select obj1 as field1,
+ *        BuildSamzaSqlRelRecord("field21", obj21, "field22", obj22) as field2,
```

Comment:
Should the java-doc reflect the UDF name? BuildSamzaSqlRelRecord vs BuildOutputRecordUdf